enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,266 @@
+
+### OCP Build & Release
+
+Phase 0 will involve building and publishing of [nightly builds][OCPPLAN-7640]. The release payload will be a manifest listed image with the component images being manifest listed as well. All 4 architectures - x86, ARM, Power and Z will be included in the manifestlist. Existing architecture specific release payloads continue to be delivered through their individual
+release controllers. A new heterogeneous release controller is bootstrapped and displays "4-stable" entries for ART constructed release payloads.
+
+`oc adm release ...` verbs would be [enhanced][WRKLDS-370] to allow the creation and management of manifestlist release payloads. Sub-verbs like `mirror`, `info`, and `extract` must be adapted to support manifestlist based payloads.
+
+In the case of `oc adm release new`, the manifestlist flow would be triggered whenever the CVO image in a source image/imagestream is a manifestlist. If the source CVO image is a standard manifest, the generated release payload will also be a manifest. If the source CVO image is a manifestlist, the generated release
+payload would be a manifestlist(containing a manifest for each architecture possessed by the CVO manifestlist). In either case, oc adm release new would permit non-CVO component images to be manifest or manifestlists and pass them through directly to the resultant release manifest(s). If a manifestlist release payload is generated, each architecture specific
+release payload manifest will reference the same pullspecs provided in the input imagestream.
+
+Since the internal Openshift registry does not support manifestlists, the release controller would use quay.io. There will be a new architecture option added to Cincinnati called "multi".
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for heterogeneous payloads which CVO will need to look at to make upgrade recommendations. There are a few upgrade scenarios to consider with the addition of heterogeneous payloads:
+
+- heterogeneous -> heterogeneous payload upgrade
+- homogeneous -> heterogeneous payload upgrade
+- heterogeneous -> homogeneous payload upgrade
+
+The support for homogeneous payloads will exist so customers running disconnected homogeneous clusters do not incur the cost of mirroring the additional architectures if they were to use heterogeneous payloads. This means that in order for CVO to find the current release in the graph, it will have to know about the existing homogeneous graph as well in addition
+to the new heterogeneous graph. There are few options on the best way to make update recommendations especially in the case of upgrading from a homogeneous to a heterogeneous payload:
```

Comment:
i don't know how desirable it is(question for @sdodson ) but another approach is to just introduce an entirely new channel (e.g. stable-4.11-multiarch).

If you want to go from homogeneous to hetereogeneous, you would switch channels.

Reply:
resolved after recent discussions .